cyipopt/tests/unit/test_scipy_optional.py

```diff
@@ -575,3 +571,16 @@ def test_minimize_late_binding_bug():
     assert res.success
     np.testing.assert_allclose(res.x, ref.x)
     np.testing.assert_allclose(res.fun, ref.fun)
+
+@pytest.mark.skipif("scipy" not in sys.modules,
+                    reason="Test only valid if Scipy available.")
+def test_minimize_callback():
+    # Test that `callback` works
+    def f(x):
+        return x @ x
+
+    def callback(intermediate_result):
+        assert intermediate_result.fun == f(intermediate_result.x)
+        assert False
```

Comment:
This was added just to make sure that the `callback` function is indeed being called and the previous assertion is passing. Surprisingly, CI does not record a failure, even though `callback` is definitely being executed. We get this warning:

```
cyipopt/tests/unit/test_scipy_optional.py::test_minimize_callback
  /usr/share/miniconda3/envs/test-environment/lib/python3.11/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: 'ipopt_wrapper.intermediate_cb'
  
  Traceback (most recent call last):
    File "/home/runner/work/cyipopt/cyipopt/cyipopt/scipy_interface.py", line 255, in intermediate
      iterate = self.get_current_iterate()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "cyipopt/cython/ipopt_wrapper.pyx", line 701, in ipopt_wrapper.Problem.get_current_iterate
      raise RuntimeError(
  RuntimeError: get_current_iterate only supports Ipopt version >=3.14.0 CyIpopt is compiled with version 3.13.4
  
    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
```

1. Why is this not tracked as a failure?
2. Can we compile with Ipopt version 3.14.0?

Reply:
If you try to use get_current_iterate() with ipopt < 3.14 you will get an error (as designed). We technically support ipopt 3.11+ (since 3.11 is what is available in Debian/Ubuntu).